### DISCUSSION TITLE
link paths overlap

### DISCUSSION THREAD

**Discussion by anxiao666** [clientIO/joint#2191] [Q&A]
### Introduction

I use a manhattan router to connect the ports of the two elements，When I move the element, the path overlapping problem occurs，I want the paths not to overlap，is there any good way

![企业微信截图_16850874121567](https://github.com/clientIO/joint/assets/50686445/36f22f2f-275a-4d72-bd58-13d1bf13f088)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by anxiao666** [comment]
Is there no way?

**Comment by kumilingus** [comment]
Not with the `manhattan` router. You need to:

- write a custom router tailored to your needs:
  See [example](https://jsfiddle.net/kumilingus/jvdn451y/).

- use an automatic layout (e.g. [ELKJS](https://www.jointjs.com/demos/eclipse-kernel-layout), [MSAGL](https://microsoft.github.io/msagljs/)) that routes your links so they do not overlap.
   As far as I know there is no standalone router available (that would take the position of elements and route only the links).
   See [ELKJS issue](https://github.com/eclipse/elk/issues/315).
